### PR TITLE
Set up clippy for a number of crates + rename 15-wires -> kimchi

### PR DESCRIPTION
**Pull request by mimoo**
In this PR I ran:

```sh
$ cargo +nightly clippy --fix
```

and enforced clippy in CI on some crates. We can't enforce clippy for all crates without doing a lot of manual fixes, so this is better than nothing for now. Long-term we should fix all the warnings for `dlog/commitment`, `circuits/kimchi` and `dlog/kimchi` at the very least

I also moved:

```
proof-systems/
├── circuits/
│   └── 15-wires -> kimchi
├── dlog/ 
│   └── 15-wires -> kimchi
```

### CODE REVIEW

**Comment by mimoo** [comment]
perhaps I'll wait for @imeckler plookup unification to land before landing this one. 

Also, it'd be nice to add the rest of https://github.com/o1-labs/proof-systems/pull/108/files

**Review by mrmr1993 — APPROVED** [review]

**Comment by mimoo** [comment]
rebased

**Pull request merged**
